src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala

```diff
@@ -82,12 +82,32 @@ case class DepositIngestTask(deposit: Deposit,
       _ = debug(s"Assigning curator role to ${deposit.depositorUserId}")
       _ <- instance.dataset(persistentId).assignRole(RoleAssignment(s"@${deposit.depositorUserId}", DefaultRole.curator.toString))
       _ <- instance.dataset(persistentId).awaitUnlock()
-      _ <- if (publish) publishDataset(persistentId)
+      publicationDate <- getPublicationdate(optAmd)
+      _ <- if (publish) publishDataset(persistentId, publicationDate)
            else keepOnDraft()
     } yield ()
     // TODO: delete draft if something went wrong
   }
 
+  def getPublicationdate(optAmd: Option[Node]): Try[String] = Try {
+    optAmd
+      .flatMap(amd => findOldestChangeDate(amd))
+      .getOrElse(LocalDate.now().toString)
```

Comment:
I think we should use the normal publish API for non-migration deposits.

Reply:
It was my suggestion. Deposit date in emd/ddm is not the date it became public, at least not at EASY. You are saying this is meant for future deposits going straight into dataverse? I was not aware of that context.